### PR TITLE
Integration testing for /lib/input/check_headers.py

### PR DESCRIPTION
**Pull request by tejalwakchoure**
## Description
 -Separated input and output files in different directories for smoother execution.

 ## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notify reviewers
@prasadtalasila @achyudhk @hrishikesh-dahiya

### CODE REVIEW

File: lib/input/check_headers.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def check_validity(self, check_unavailable_uid='False', json_header_filename='he
 		return self.last_uid_read
 
 
-	def remove_unwanted_headers(self, to_remove=unwanted_uid, json_header_filename='headers.json'):
+	def remove_unwanted_headers(self, to_remove=unwanted_uid, in_json_header_filename="headers.json", out_json_header_filename="headers.json"):
```
Comment (prasadtalasila) [inline lib/input/check_headers.py]: two variable names that differ only in prefix are confusing. Please use one variable with variable name as `json_headers`.

File: lib/input/check_headers.py [hunk 2]
```diff
@@ -140,19 +140,19 @@ def remove_unwanted_headers(self, to_remove=unwanted_uid, json_header_filename='
...
 
-			with open(json_header_filename, 'w') as json_file:
+			with open(out_json_header_filename, 'w') as json_file:
 				for json_obj in write_to_file:
 					json.dump(json_obj, json_file, indent=1)
 					json_file.write("\n")
 
 
-	def remove_duplicate_headers(self,to_remove=duplicate_uid, json_header_filename='headers.json'):
+	def remove_duplicate_headers(self,to_remove=duplicate_uid, in_json_header_filename="headers.json", out_json_header_filename="headers.json"):
```
Comment (prasadtalasila) [inline lib/input/check_headers.py]: same advice. Please keep a single variable name.

File: test/integration_test/lib/input/test_check_headers.py [hunk 3]
```diff
@@ -7,9 +7,10 @@
 
 class TestCheckHeaders(object):	
 	
-	headers_file='./test/integration_test/data/headers_for_check.json'
+	in_headers_file='./test/integration_test/data/headers_for_check.json'
 	unwanted_uid_file='./test/integration_test/data/unwanted_uid.txt'
 	uid_map_file='./test/integration_test/data/thread_uid_map.json'
+	out_headers_file='./etc/.tmp/integration_test/lib/input/check_headers/check_headers_output.json'
```
Comment (prasadtalasila) [inline test/integration_test/lib/input/test_check_headers.py]: this van be `expected_headers_file` and `in_headers_file` can be replaced with `headers_file`

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@tejalwakchoure please see the suggested changes.

File: lib/input/check_headers.py [hunk 4]
```diff
@@ -125,14 +125,15 @@ def check_validity(self, check_unavailable_uid='False', json_header_filename='he
 		return self.last_uid_read
 
 
-	def remove_unwanted_headers(self, to_remove=unwanted_uid, json_header_filename='headers.json'):
+	def remove_unwanted_headers(self, to_remove=unwanted_uid, json_header_filename="headers.json", output_filename="headers.json"):
```
Comment (prasadtalasila) [inline lib/input/check_headers.py]: In the previous signature `json_header_filename` was being used to read input and update the output. We can use the same model to simplify the number of parameters. Also `json_headers` is a better name than `json_header_filename`. Please consider making the change for other functions as well.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]

File: lib/input/check_headers.py (hunk 4 above)
Comment (tejalwakchoure) [reply lib/input/check_headers.py]: @prasadtalasila The issue with using the same parameter for input and updated output is that the file changes after every execution and hence gives errors while testing.

Comment (prasadtalasila) [reply lib/input/check_headers.py]: Makes sense. Please go ahead with your convention.

**Review by prasadtalasila — APPROVED** [review]

**Pull request merged**